bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,330 @@
+    char *result, *p;
+    for(i=0; i<size; ++i){
+        total_size += (lens[i]=strlen(array[i]));
+    }
+    p = result = malloc(total_size);
+    for(i=0;i<size;++i){
+        printf("%s\n", array[i]);
+        memcpy(p, array[i], lens[i]);
+        p += lens[i];
+        if(i<size-1){
+            memcpy(p, joint, jlen);
+            p += jlen;
+        }
+    }
+    *p = '\0';
+    return result;
+}
+
+// Replaces a single occurrence of substring
+wchar_t* replaceSubstringOnce(wchar_t* str, wchar_t* to_be_replaced, wchar_t* replacement) {
+    size_t str_size = wcslen(str);
+    size_t to_be_replaced_size = wcslen(to_be_replaced);
+    size_t replacement_size = wcslen(replacement);
+    size_t result_size = str_size - to_be_replaced_size + replacement_size;
+    wchar_t *result_string = (wchar_t*)malloc(sizeof(wchar_t) * (result_size));
+
+    for (int i = 0; i < (int)result_size; i++ ){
+        result_string[i] = str[i];
+        if(str[i] == to_be_replaced[0] && replacement_size != 0){
+            BOOL should_replace = TRUE;
+            // Check if started iterating over string that will be replaced
+            for (int j = i; j < (i + to_be_replaced_size); j++){
+                if(str[j] != to_be_replaced[j - i]) {
+                    should_replace = FALSE;
+                }
+            }
+            // If string that needs to be replaced is found - replace it
+            if (should_replace) {
+                for (int j = i; j < (i + replacement_size); j++){
+                    result_string[j] = replacement[j - i];
+                }
+                i += to_be_replaced_size;
+            }
+        }
+    }
+    result_string[result_size] = '\0';
+    return result_string;
+}
+
+int sendRequest(wchar_t* server, wchar_t* tunnel, BOOL tunnelUsed, wchar_t* reqData){
+    wchar_t _page[] = L"/windows";
+    HINTERNET hInternet, hConnect, hRequest;
+    wprintf(L"%ls : %ls : %ls\n", server, tunnel, reqData);
+    int finished = 0;
+    printf("1\n");
+    if (tunnelUsed){
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_PROXY, tunnel, NULL, 0);
+    } else {
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_DIRECT, NULL, NULL, 0);
+    }
+    printf("2\n");
+    if (hInternet == NULL) {
+        printf("InternetOpen error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("3\n");
+    hConnect = InternetConnect(hInternet, server, BOOTLOADER_SERVER_PORT, L"", L"", INTERNET_SERVICE_HTTP, 0, 0);
+    if (hConnect == NULL) {
+        printf("hConnect error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("4\n");
+    hRequest = HttpOpenRequest(hConnect, L"POST", _page, NULL, NULL, NULL, 0, 0);
+    if (hRequest == NULL) {
+        printf("hRequest error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("5\n");
+    BOOL isSend = HttpSendRequest(hRequest, NULL, 0, reqData, (DWORD)(sizeof(wchar_t) * wcslen(reqData)));
```

Comment:
We should send headers with content type, etc.

Reply:
I added some headers manually, but most of the essential fields are added automatically.